### PR TITLE
code coverage

### PR DESCRIPTION
**Pull request by webb04**
## Why are you doing this?

It's part of our health OKR to have 75% test coverage on our core parsing logic.

-----

To run: `npm run test`

I've added `renderer.ts` but we can include more files with parsing logic.

The threshold is currently 15% so that the builds don't fail but we should increase this as we add tests.

### CODE REVIEW

File: src/renderer.test.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { renderAll } from './renderer';
+
+describe('', () => {
+    test('', () => {
+        expect(renderAll("")).toBeDefined();
```
[the remark this answers is not in the document]
Comment (webb04) [reply src/renderer.test.ts]: Yeah this test should be changed but I expected that to be recognised as covered code, but it's not... it's still reporting as 0% coverage and the `renderAll` line is displayed as `Uncovered`

File: package.json [hunk 2]
```diff
@@ -20,6 +21,34 @@
...
+    "collectCoverage": true,
+    "transform": {
+      "^.+\\.tsx?$": "ts-jest"
+    },
+    "coverageThreshold": {
+      "global": {
+        "branches": 15,
+        "functions": 15,
+        "lines": 15,
+        "statements": 15
```
Comment (JamieB-gu) [inline package.json]: 15% of each?

File: package.json [hunk 3]
```diff
@@ -20,6 +21,34 @@
...
+      }
+    },
+    "coverageReporters": [
+      "json",
+      "lcov",
+      "text",
+      "clover",
```
Comment (JamieB-gu) [inline package.json]: What's clover?

File: src/components/shared/keyline.test.tsx [hunk 4]
```diff
@@ -6,24 +6,28 @@ import { Layout } from 'article';
 
 configure({ adapter: new Adapter() });
 
+const styles = (component: any): string => component.props().css
+    .map(({ styles }: any) => styles)
```
Comment (JamieB-gu) [inline src/components/shared/keyline.test.tsx]: Do we have a stricter type for these?

File: src/renderer.test.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import { makeCaption } from './renderer';
+
+describe('renderer returns expected content', () => {
+    test('formats captions', () => {
```
Comment (JamieB-gu) [inline src/renderer.test.ts]: Should we be testing functions that are private to a module?

File: tsconfig.json [hunk 6]
```diff
@@ -6,7 +6,7 @@
         // Import non-ES modules as default imports.
         "esModuleInterop": true,
         // Don't transpile jsx (handled by babel w/ emotion)
-        "jsx": "preserve",
+        "jsx": "react",
```
Comment (JamieB-gu) [inline tsconfig.json]: What does this mean?

File: package.json (hunk 2 above)
Comment (webb04) [reply package.json]: Should be 75%+ eventually but I set it lower so the builds don't fail

File: tsconfig.json (hunk 6 above)
Comment (webb04) [reply tsconfig.json]: This is to get it to work with `ts-jest`. Client and server configs still use `preserve`

File: package.json (hunk 3 above)
Comment (webb04) [reply package.json]: https://istanbul.js.org/docs/advanced/alternative-reporters/

Comment (webb04) [reply package.json]: Removed some of them and kept `text` and `html` which you can see by running `open coverage/index.html`

**Pull request merged**